odl/space/fspace.py

```diff
@@ -500,6 +500,11 @@ def element(self, fcall=None, vectorized=True):
 
             return self.element_type(self, fcall)
 
+    @property
+    def out_dtype(self):
+        """Output data type of functions in this space."""
+        return 'complex128' if self.field == ComplexNumbers() else 'float64'
```

Comment:
So now this is fully customizable. In `FunctionSet` [you can give the output data type](https://github.com/odlgroup/odl/blob/6fbc2e3cbb8f0d239aa295806ec3c9b12ae99e32/odl/space/fspace.py#L84-L90), and if you don't, the ["lazy" approach](https://github.com/odlgroup/odl/blob/6fbc2e3cbb8f0d239aa295806ec3c9b12ae99e32/odl/space/fspace.py#L62-L64) is used.
For `FunctionSpace`, again, you can give a data type explicitly, but if you don't, the `field` argument [is used to determine the default](https://github.com/odlgroup/odl/blob/6fbc2e3cbb8f0d239aa295806ec3c9b12ae99e32/odl/space/fspace.py#L484-L518) (`'float64'` if both are `None`). Then there are some consistency checks.
If the field is not `RealNumbers` or `ComplexNumbers`, you need to give the data type explicitly. This could be changed to setting it to `None` as for `FunctionSet` and rely on lazy evaluation, now that I think of it.